### DISCUSSION TITLE
progressive webapp

### DISCUSSION THREAD

**Discussion by bjornclauw** [Budibase/budibase#5668] [Ideas]
Hi,

just wondering if there were any plans on adding a manifest.json to the frontend so the apps can be installed as progressive webapps on android.

workaround for me now is custom webview (android) app that opens the budibase app url.

**Comment by shogunpurple** [comment]
Hi @bjornclauw - this is something we have planned to do for a while! We just haven't got round to it yet. I'm going to convert this one into an issue and we will try to address it soon. What would you like to see from this feature? The idea I had originally for it was to add some UI into the builder, where you can "turn on" progressive web app features, and update the keys in the `manifest.json` from within the builder, then your app will ship with it. How does that sound?

**Comment by bjornclauw** [comment]
Hi @shogunpurple, that sounds pretty sweet. 

I am using BB almost exclusively on mobile. I really like how the app scales to mobile and how Spectrum looks. (hobby user, industrial maintenance engineer -> extensively using API calls to acces big databases who otherwise would by too cluncky to manipulate on mobile, crosslinking to data between databases, retrieving documents, images, sending POST commands, ... ).

_Below are some of the things I harcoded in my custom android app, i don't know if it would be possible to implement into BB._

- Possible to somehow create your own 'sticky header menu'/customize existing one?
- Add option 'close webapp' to the button/menu (links) component.
- Add the option 'open mobile menu flyout' to the button component.
- Had some various issues getting the 'add attachment' component working. (acces rights for camera, file acces)
- QR code component would be great (solved trough custom api for the moment)
- Had some trouble with cookies and linking image component to 'protected' images on different domain.

![example](https://user-images.githubusercontent.com/32811735/166149000-9e5276f8-bbe1-4224-a049-2dfe9a16ff4b.jpg)

Happy to assist in any way i can.

**Comment by pakin1** [comment]
Is there any way at this moment to make my app installable?

**Comment by estelle-d** [comment]
Hello, 
Any news on that feature?
